velodyne_driver/src/lib/input.cc

```diff
@@ -90,55 +97,57 @@ namespace velodyne_driver
    *  @param private_nh ROS private handle for calling node.
    *  @param port UDP port number
    */
-  InputSocket::InputSocket(ros::NodeHandle private_nh, uint16_t port):
+  InputSocket::InputSocket(rclcpp::Node * private_nh, uint16_t port):
     Input(private_nh, port)
   {
     sockfd_ = -1;
-    
-    if (!devip_str_.empty()) {
-      inet_aton(devip_str_.c_str(),&devip_);
-    }    
+
+    if (!devip_str_.empty())
+      {
+        ::inet_aton(devip_str_.c_str(), &devip_);
+      }
 
     // connect to Velodyne UDP port
-    ROS_INFO_STREAM("Opening UDP socket: port " << port);
-    sockfd_ = socket(PF_INET, SOCK_DGRAM, 0);
+    RCLCPP_INFO(private_nh->get_logger(), "Opening UDP socket: port %u", port);
+
+    sockfd_ = ::socket(PF_INET, SOCK_DGRAM, 0);
     if (sockfd_ == -1)
       {
-        perror("socket");               // TODO: ROS_ERROR errno
+        RCLCPP_ERROR(private_nh->get_logger(), "Error opening socket: %s", ::strerror(errno));
         return;
       }
-  
+
     sockaddr_in my_addr;                     // my address information
     memset(&my_addr, 0, sizeof(my_addr));    // initialize to zeros
     my_addr.sin_family = AF_INET;            // host byte order
-    my_addr.sin_port = htons(port);          // port in network byte order
+    my_addr.sin_port = ::htons(port);          // port in network byte order
```

Comment:
Hm not sure this will work, since `htons` is a macro (at least on MacOS). I have been trying to compile it on Mac and it doesn't like this. (https://stackoverflow.com/questions/15002550/c-macros-and-namespaces)

Reply:
How about:

    #undef htons;
    my_addr.sin_port = htons(port);

Since this should only affect the code from this point to the end of this file during compilation, it shouldn't be an issue.

Reply:
It's failing CI as well on this item.

Reply:
Ah, I didn't realize it was a macro on some platforms.  Will revert back to just `htons`.